include/motor.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef MOTOR_H
+#define MOTOR_H
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+//#include "freertos/semphr.h"
+
+/**
+ *  Motor
+ *   
+ *  position = mm
+ *  speed = mm/s
+ *  acceleration = mm/s^2
+ */
+
+#define DEBUG_CLIPPING              // Show debug messages when motions violating the machine 
+                                    // physics are commanded
+
+// If start > end, indicates an inversion of direction
+typedef struct {
+  float start;
+  float end;
+  float keepout; /*  Soft endstop preventing hard crashes in mm. Will be 
+                  *  subtracted twice from physicalTravel. Should be 
+                  *  sufficiently to completley drive clear from 
+                  *  homing switch */
+} motionBounds;
+
+#define MOTOR_FLAG_ENABLED (1 << 0)
+
+#define MOTOR_FLAG_WARNING (1 << 1)
+#define MOTOR_FLAG_ERROR (1 << 2)
+#define MOTOR_FLAG_FATAL (1 << 3)
+
+#define MOTOR_FLAG_AT_TARGET (1 << 4)
+#define MOTOR_FLAG_MOTION_ACTIVE (1 << 5)
+#define MOTOR_FLAG_HOMED (1 << 6)
+
+#define MOTOR_FLAG_OVERHEATING (1 << 7)
+#define MOTOR_FLAG_OVERFORCE (1 << 8)
+
+typedef enum {
+  INACTIVE = 0,
+  ACTIVE = 1,
+  HOMING = 2, 
+  ERROR = 3
+} MotorState;
+
+class MotorInterface {
+  public:
+    void enable() { this->addStatusFlag(MOTOR_FLAG_ENABLED); }
+    // TODO - virtual void disable();
+    
+    // Safety Bounds
+    void setMaxSpeed(float speed) { this->maxSpeed = speed; }
+    float getMaxSpeed() { return this->maxSpeed; }
+
+    void setMaxAcceleration(float acceleration) { this->maxAcceleration = acceleration; }
+    float getMaxAcceleration() { return this->maxAcceleration; }
+
+    void setBounds(motionBounds bounds) { this->bounds = bounds; }
+    motionBounds getBounds() { return this->bounds; }
```

Comment:
Are bounds even being used by anything currently? I need to add clipping protection to prevent movement outside of bounds. Can add it into the safe goToPos function so all motors have the safety built-in

Reply:
Should be generalized in motor.hpp